python/smqtk/web/nn_descriptor_service/__init__.py

```diff
@@ -0,0 +1,230 @@
+
+        :param json_config: JSON configuration dictionary
+        :type json_config: dict
+
+        """
+        super(NearestNeighborServiceServer, self).__init__(json_config)
+
+        # Descriptor factory setup
+        self.log.info("Initializing DescriptorElementFactory")
+        self.descr_elem_factory = DescriptorElementFactory.from_config(
+            self.json_config['descriptor_factory']
+        )
+
+        # Descriptor generator configuration labels
+        #: :type: dict[str, dict]
+        self.generator_config = self.json_config['descriptor_generator']
+
+        self.nn_index = plugin.from_plugin_config(
+            json_config['nn_index'],
+            get_nn_index_impls
+        )
+
+        self.descriptor_generator_inst = plugin.from_plugin_config(
+                                            self.generator_config,
+                                            get_descriptor_generator_impls)
+
+
+        @self.route("/nn/<path:uri>")
+        def compute_nearest_neighbors(uri):
+            """
+            Data modes for upload/use::
+
+                - local filepath
+                - base64
+                - http/s URL
+
+            The following sub-sections detail how different URI's can be used.
+
+            Local Filepath
+            --------------
+
+            The URI string must be prefixed with ``file://``, followed by the
+            full path to the data file to describe.
+
+            Base 64 data
+            ------------
+
+            The URI string must be prefixed with "base64://", followed by the
+            base64 encoded string. This mode also requires an additional
+            ``?content_type=`` to provide data content type information. This
+            mode saves the encoded data to temporary file for processing.
+
+            HTTP/S address
+            --------------
+
+            This is the default mode when the URI prefix is none of the above.
+            This uses the requests module to locally download a data file
+            for processing.
+
+            JSON Return format::
+                {
+                    "success": <bool>
+
+                    "message": <str>
+
+                    "neighbors": <None|list[float]>
+
+                    "reference_uri": <str>
+                }
+
+            :type descriptor_label: str
+            :type uri: str
+
+            """
+            message = "execution nominal"
+            descriptor = None
+
+            de = None
+            try:
+                de = self.resolve_data_element(uri)
+            except ValueError, ex:
+                message = "URI resolution issue: %s" % str(ex)
+
+            if de:
+                try:
+                    descriptor = self.descriptor_generator_inst.\
+                        compute_descriptor(de, self.descr_elem_factory)
+
+                    if descriptor is not None:
+                        descriptor.set_vector(descriptor.vector().flatten())
```

Comment:
Right and that was the TODO I had earlier which I removed as I thought you looked at it. Yes, I was getting (4096, 1, 1) both when building and using the index and since flann was not liking it. I debug further but I could not find any solutions to the problem other than flattening it. You can see in my build index example, I am doing the same.